lib/bson/binary.rb

```diff
@@ -207,7 +207,8 @@ def to_uuid(representation = nil)
         if representation && representation != :standard
           raise ArgumentError, "Binary of type :uuid can only be stringified to :standard representation, requested: #{representation.inspect}"
         end
-        data.split('').map { |n| '%02x' % n.ord }.join.sub(/(.{8})(.{4})(.{4})(.{12})/, '\1-\2-\3-\4')
+
+        data.split('').map { |n| '%02x' % n.ord }.join.sub(/(.{8})(.{4})(.{4})(.{4})(.{12})/, '\1-\2-\3-\4-\5')
```

Comment:
To guard against this type of issue in the future, can you please add `\A` and `\z` anchors to the regular expressions changed by this PR?

Reply:
Added the anchors